src/store/Account.ts

```diff
@@ -511,39 +513,30 @@ export default {
         { root: true },
       )
 
-      try {
-        // - prepare REST parameters
-        const repositoryFactory = rootGetters['network/repositoryFactory'] as RepositoryFactory
-        const transactionHttp = repositoryFactory.createTransactionRepository()
-
-        // - prepare scoped *confirmation listener*
-        const listener = repositoryFactory.createListener()
-        await listener.open()
-
-        // - announce hash lock transaction and await confirmation
-        transactionHttp.announce(signedLock)
-
-        // - listen for hash lock confirmation
-        return new Promise((resolve, reject) => {
-          const address = Address.createFromRawAddress(issuer)
-          return listener.confirmed(address).subscribe(
-            async () => {
-              // - hash lock confirmed, now announce partial
-              await transactionHttp.announceAggregateBonded(signedPartial)
-              commit('removeSignedTransaction', signedLock)
-              commit('removeSignedTransaction', signedPartial)
-              return resolve(new BroadcastResult(signedPartial, true))
-            },
-            () => {
-              commit('removeSignedTransaction', signedLock)
-              commit('removeSignedTransaction', signedPartial)
-              reject(new BroadcastResult(signedPartial, false))
-            },
-          )
-        })
-      } catch (e) {
-        return new BroadcastResult(signedPartial, false, e.toString())
-      }
+      // - prepare REST parameters
+      const repositoryFactory = rootGetters['network/repositoryFactory'] as RepositoryFactory
+      const transactionHttp = repositoryFactory.createTransactionRepository()
+      const receiptsHttp = repositoryFactory.createReceiptRepository()
+      const transactionService = new TransactionService(transactionHttp, receiptsHttp)
+      // - prepare scoped *confirmation listener*
+      const listener = repositoryFactory.createListener()
+      await listener.open()
+
+      return transactionService
+        .announceHashLockAggregateBonded(signedLock, signedPartial, listener)
+        .pipe(
+          catchError((error) => of(new BroadcastResult(signedPartial, false, error))),
+          mapTo(new BroadcastResult(signedPartial, true)),
+          timeoutWith(
+            10_000_000, // 10 minutes timeout
```

Comment:
Seems like a very long period to be inspecting blocks ; Wouldn't 2 minutes suffice? If there is no block including the hash lock after 2 minutes, there is already something wrong, that's what i mean